src/main/java/org/lisoft/lsml/view_fx/util/WeaponSummary.java

```diff
@@ -151,7 +151,7 @@ protected double computeValue() {
                 if (weapon.getDamagePerProjectile() == 0) {
                     return 0.0;
                 }
-                return ammoRounds.get() * weapon.getDamagePerShot() / weapon.getAmmoPerPerShot();
+                return ammoRounds.get() * weapon.getDamagePerShot() / weapon.getRoundsPerShot();
```

Comment:
From memory, this should become:

```suggestion
                if (weapon instanceof AmmoWeapon) {
                    AmmoWeapon ammoWeapon = (AmmoWeapon) weapon;
                    return ammoRounds.get() * ammoWeapon .getDamagePerShot() / ammoWeapon .getAmmoPerShot();
                }
                return Double.POSITIVE_INFINITY;   
```

`ammoRounds.get()` would return `Double.POSITIVE_INFINITY` for non `AmmoWeapon` types so the above computation would result in `Double.POSITIVE_INFINITY * 10 / 1` for ER PPC for example.  I.e. the logic was: energy weapons have infinite ammo so the result becomes infinity.

Reply:
thanks. I thought there would be an easy way.

Reply:
pushed it.